src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -270,6 +285,29 @@ std::pair<int, int> AdaptDispatch::_GetVerticalMargins(const til::rect& viewport
     return { topMargin, bottomMargin };
 }
 
+// Routine Description:
+// - Returns the coordinates of the horizontal scroll margins.
+// Arguments:
+// - bufferWidth - The width of the buffer
+// Return Value:
+// - A std::pair containing the left and right coordinates (inclusive).
+std::pair<int, int> AdaptDispatch::_GetHorizontalMargins(const til::CoordType bufferWidth) noexcept
```

Comment:
i'm slightly alarmed that a `_get` function changes state, but I'll circle back after I read the rest of adaptDispatch

Reply:
okay, `GetVerticalMargins` does as well. i'm less worried

Reply:
In the long term I was hoping there'd be a way for `AdaptDispatch` to register a callback where it's notified of buffer and window size changes, and that's when it would recalculate the margin boundaries. If we could do that, then these `_Get*Margin` methods would just be returning precalculated values.

Reply:
As simple as this function is, I personally don't think caching is necessary. Just eyeballing it, I suspect it runs in ~10ns on modern CPUs, or the equivalent of doing a 2 `call`s. 😅

Reply:
It's not just the function itself, though. The vertical margin calculation depends on the viewport size, which we need to retrieve with a virtual function call, and the horizontal margin calculation depends on the buffer size, which means another virtual call to obtain the buffer. Generally speaking we're going to need the buffer anyway, so that's not really extra, but ideally I'd like to have eliminated all of those calls if possible.

You'll know better than me how much of difference that would make, so if you're happy with it how it is then that's good enough for me. I was just concerned that I might be introducing a bunch of overhead with this stuff which would undermine all the work you've been doing to improve the performance.

Reply:
That's a good point! One of the big refactors I'm planning to do next in this area will be to make `IRenderData` less virtual and to expose the text buffer, viewport, etc. as a plain struct members, to reduce the overhead of virtual calls in our hot paths. So if all goes well, we'll solve that issue categorically soon. 🙂